R/add_risktable.R

```diff
@@ -152,32 +141,59 @@ add_risktable.ggsurvfit <- function(
     },
     statlist = as.list(statlist),
     title = as.list(title)
-  )
+    )
 
-# Align plot and table by adjusting width ---------------------------------
+  # Align plot and table by adjusting width ---------------------------------
 
   gglist <- list(gg) %>%
     base::append(tbls)
-  
+
   ggA <- gglist %>%
     align_plots()
-  
-# Create plot and add class -----------------------------------------------
+
+  # Create plot and add class -----------------------------------------------
 
   ## cowplot allows to align according to an axis (+left) and change the heigth
   ggB <- cowplot::plot_grid(plotlist = ggA,
                             align = "none",
                             nrow = length(ggA),
                             rel_heights = c(1-(8/50 * (length(ggA)-1)), rep(8/50, length(ggA)-1))
-                          )
-  
-  class(ggB) <- c(class(ggB), "ggsurvfit")
+  )
+
+  class(ggB) <- c(class(ggB), intersect(class(gg), c("ggsurvfit", "ggtidycmprsk")))
+
+  # Add individual components -----------------------------------------------
 
-# Add individual components -----------------------------------------------
-  
   components <- append(list(gg), tbls)
   names(components) = c("visR_plot", title)
   ggB[["components"]] <- components
-  
+
   return(ggB)
 }
+
+#' @rdname add_risktable
+#' @export
+add_risktable.ggtidycuminc <- add_risktable.ggsurvfit
+
+
+# this function extracts the survift or tidycmprsk object from the ggplot objects
+.extract_estimate_object <- function(gg) {
```

Comment:
Can you put this utility function in utils_visR file please?

Reply:
Done! ✔️